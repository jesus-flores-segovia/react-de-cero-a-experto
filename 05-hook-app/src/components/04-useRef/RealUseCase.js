import React, {useState} from 'react';
import { MultipleCustomHooks } from '../03-examples/MultipleCustomHooks';

import '../02-useEffect/effects.css';

export const RealUseCase = () => {

    const [show, setShow] = useState(false);

    return (
        <div>
            <h1>RealUseCase</h1>
            {show && <MultipleCustomHooks/>}
            <button className="btn btn-primary mt-5" onClick={() => setShow(!show)}>Show/Hide</button>
        </div>
    )
}