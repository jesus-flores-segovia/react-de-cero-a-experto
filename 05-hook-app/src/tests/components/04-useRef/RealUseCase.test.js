import React from 'react';
import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import {RealUseCase} from "../../../components/04-useRef/RealUseCase";

describe("Tests inside file RealUseCase.test.js", () => {

    test("'RealUseCase' component must show correctly", () => {
        const wrapper = shallow(<RealUseCase/>)
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find("MultipleCustomHooks").exists()).toBe(false);
    });

    test("'RealUseCase' component must show 'MultipleCustomHooks' component", () => {
        const wrapper = shallow(<RealUseCase/>)

        wrapper.find("button").simulate("click");

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find("MultipleCustomHooks").exists()).toBe(true);
    });
});