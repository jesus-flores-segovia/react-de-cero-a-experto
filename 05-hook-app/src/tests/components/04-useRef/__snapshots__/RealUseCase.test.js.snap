// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests inside file RealUseCase.test.js 'RealUseCase' component must show 'MultipleCustomHooks' component 1`] = `
<div>
  <h1>
    RealUseCase
  </h1>
  <MultipleCustomHooks />
  <button
    className="btn btn-primary mt-5"
    onClick={[Function]}
  >
    Show/Hide
  </button>
</div>
`;

exports[`Tests inside file RealUseCase.test.js 'RealUseCase' component must show correctly 1`] = `
<div>
  <h1>
    RealUseCase
  </h1>
  <button
    className="btn btn-primary mt-5"
    onClick={[Function]}
  >
    Show/Hide
  </button>
</div>
`;
